app/public/src/game/components/board-manager.ts

```diff
@@ -68,6 +69,7 @@ export default class BoardManager {
     if (pokemonUI) {
       pokemonUI.destroy(true)
     }
+    this.pokemons.delete(pokemonToRemove.id)
```

Comment:
Bugfix: the board map was not cleared when a pokemon was removed. This led to ghost pokemons being kept in the board map whenever they are removed, when changing the player observed or when evolving a pokemon. @keldaan-ag  you should check again but I'm pretty sure this was not intended and could be responsible of other bugs I'm not aware of.